exonum/src/api/websocket.rs

```diff
@@ -0,0 +1,157 @@
+// Copyright 2018 The Exonum Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! WebSocket API.
+
+use actix::*;
+use actix_web::ws;
+
+use rand::{self, Rng, ThreadRng};
+
+use std::cell::RefCell;
+use std::collections::HashMap;
+
+use api::ServiceApiState;
+use crypto::Hash;
+
+/// WebSocket Message for communication between Clients(`WsSession`) and Server(`WsServer`).
+#[derive(Message, Debug)]
+pub(crate) struct Message(pub String);
+
+#[derive(Message)]
+#[rtype(usize)]
+pub(crate) struct Subscribe {
+    pub address: Recipient<Syn, Message>,
+}
+
+#[derive(Message)]
+pub(crate) struct Unsubscribe {
+    pub id: usize,
+}
+
+#[derive(Message)]
+pub(crate) struct Broadcast {
+    pub block_hash: Hash,
+}
+
+/// WebSocket server.
+pub(crate) struct WsServer {
```

Comment:
I agree with @vldm, we can name it `Server`, without `Ws` part. The same is true for other types